blog/test.py

```diff
@@ -174,16 +175,26 @@ def test_language_detection(self):
     def test_article_appear_by_default_in_articles(self):
         Article(title='A title for an article', content='<p>content</p>',
                 publication_date=datetime.datetime.utcnow(),
+                published=True,
                 author=self.user).save()
         result = self.client.get('/articles/')
         self.assertIn(b'A title for an article', result.data)
 
     def test_unpublished_article_does_not_appear_in_articles(self):
         Article(title='title', content='<p>content</p>',
+                published=False,
                 publication_date=None,
                 author=self.user).save()
         result = self.client.get('/articles/')
         self.assertNotIn(b'<article class=preview>', result.data)
+        
+    def test_published_then_unpublished_article_does_not_appear_in_articles(self):
```

Comment:
the test does not publish then unpublish.
You may mean `test_unpublished_article_does_not_appear_in_articles` ?